main/src/com/google/refine/importers/XmlImporter.java

```diff
@@ -216,7 +219,45 @@ static public class XmlParser implements TreeReader {
         static final int WHITESPACE_CHARACTERS_TOKEN = 15;
 
         public XmlParser(InputStream inputStream) throws XMLStreamException, IOException {
-            parser = createXMLStreamReader(inputStream);
+            parser = createXMLStreamReader(new InvalidXmlFilterInputStream(inputStream));
+        }
+
+        static class InvalidXmlFilterInputStream extends BufferedInputStream {
+
+            private final Pattern pattern = Pattern.compile("[^\\u0009\\u000A\\u000D\\u0020-\\uD7FF\\uE000-\\uFFFD\\u10000-\\u10FFF]+");
+
+            /**
+             * Creates a new filtered reader.
+             *
+             * @param in
+             *            an input stream object.
+             * @throws NullPointerException
+             *             if {@code in} is {@code null}
+             */
+            protected InvalidXmlFilterInputStream(@NotNull InputStream in) {
+                super(in);
+            }
+
+            @Override
+            public int read() throws IOException {
+                int c = super.read();
+                if (c == -1) return c;
+                while (pattern.matcher(String.valueOf((char) c)).matches()) {
+                    c = super.read();
+                }
+                return c;
+            }
```

Comment:
This `read()` method is inconsistent with the other one: the one above removes invalid characters, the one below replaces them by `?`. So this might lead to unstable behaviour, depending on which of the methods is used. I think both methods should have a consistent behaviour (perhaps with a slight preference for removing, since it is perhaps more likely that this produces valid XML markup).